src/pages/RFOX/components/Bridge/BridgeConfirm.tsx

```diff
@@ -0,0 +1,416 @@
+import { useAppSelector } from 'state/store'
+
+import type { RfoxBridgeQuote } from './types'
+import { BridgeRoutePaths, type BridgeRouteProps } from './types'
+
+type BridgeConfirmProps = {
+  bridgeQuote: RfoxBridgeQuote
+}
+
+const backIcon = <ArrowBackIcon />
+
+const CustomRow: React.FC<RowProps> = props => <Row fontSize='sm' fontWeight='medium' {...props} />
+
+export const BridgeConfirm: FC<BridgeRouteProps & BridgeConfirmProps> = ({ bridgeQuote }) => {
+  const toast = useToast()
+  const queryClient = useQueryClient()
+  const wallet = useWallet().state.wallet
+  const history = useHistory()
+  const translate = useTranslate()
+
+  const [approvalTxHash, setApprovalTxHash] = useState<string>()
+
+  const handleGoBack = useCallback(() => {
+    // TODO(gomes): implement me, but this should route back to stake, not bridge as there's no select route here
+    // history.push(BridgeRoutePaths.Select)
+  }, [])
+
+  const sellAsset = useAppSelector(state => selectAssetById(state, bridgeQuote.sellAssetId))
+  const buyAsset = useAppSelector(state => selectAssetById(state, bridgeQuote.buyAssetId))
+
+  const feeAsset = useAppSelector(state =>
+    selectFeeAssetByChainId(state, fromAssetId(bridgeQuote.sellAssetId).chainId),
+  )
+
+  const feeAssetMarketData = useAppSelector(state =>
+    selectMarketDataByAssetIdUserCurrency(state, feeAsset?.assetId ?? ''),
+  )
+
+  const bridgeAmountCryptoPrecision = useMemo(
+    () => fromBaseUnit(bridgeQuote.bridgeAmountCryptoBaseUnit, sellAsset?.precision ?? 0),
+    [bridgeQuote.bridgeAmountCryptoBaseUnit, sellAsset?.precision],
+  )
+
+  const sellAssetMarketDataUserCurrency = useAppSelector(state =>
+    selectMarketDataByAssetIdUserCurrency(state, bridgeQuote.sellAssetId),
+  )
+
+  const bridgeAmountUserCurrency = useMemo(
+    () =>
+      bnOrZero(bridgeAmountCryptoPrecision).times(sellAssetMarketDataUserCurrency.price).toFixed(),
+    [bridgeAmountCryptoPrecision, sellAssetMarketDataUserCurrency.price],
+  )
+
+  const accountNumberFilter = useMemo(
+    () => ({ assetId: bridgeQuote.sellAssetId, accountId: bridgeQuote.sellAssetAccountId }),
+    [bridgeQuote.sellAssetAccountId, bridgeQuote.sellAssetId],
+  )
+  const accountNumber = useAppSelector(state =>
+    selectAccountNumberByAccountId(state, accountNumberFilter),
+  )
+
+  const { data: quote, isLoading: isBridgeQuoteLoading } = useQuery({
+    ...reactQueries.swapper.arbitrumBridgeTradeQuote({
+      sellAsset: sellAsset!,
+      buyAsset: buyAsset!,
+      chainId: sellAsset!.chainId as EvmChainId,
+      sellAmountIncludingProtocolFeesCryptoBaseUnit: bridgeQuote.bridgeAmountCryptoBaseUnit,
+      affiliateBps: '0',
+      potentialAffiliateBps: '0',
+      allowMultiHop: true,
+      receiveAddress: fromAccountId(bridgeQuote.buyAssetAccountId).account,
+      sendAddress: fromAccountId(bridgeQuote.sellAssetAccountId).account,
+      accountNumber: accountNumber!,
+      wallet: wallet!,
```

Comment:
all these non-null asserts need handling

Reply:
They are, see `enabled` below - added comments and extracted to a new `isTradeQuoteQueryEnabled` for consistency in p3 bce9e368ec

Note, I didn't memoize `isTradeQuoteQueryEnabled` in preparation for `skipToken`

Reply:
Now clean and type-safety 🎉  https://github.com/shapeshift/web/pull/7020#discussion_r1631768133